eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/AbstractSynchronizedPutMultimap.java

```diff
@@ -96,4 +97,22 @@ private C getIfAbsentPutCollection(K key)
     {
         return this.map.getIfAbsentPutWith(key, this.createCollectionBlock(), this);
     }
+
+    @Override
+    public C getIfAbsentPut(K key, Iterable<? extends V> values)
+    {
+        if (Iterate.isEmpty(values))
+        {
+            return this.get(key);
+        }
+
+        C existingValue = this.getIfAbsentPutCollection(key);
+        if (existingValue.isEmpty())
+        {
+            int newSize = Iterate.addAllTo(values, existingValue).size();
+            this.addToTotalSize(newSize);
+        }
+
+        return (C) existingValue.asUnmodifiable();
+    }
```

Comment:
We don't have to grab a lock or anything? I'm concerned about the fact that we first put an empty collection into the backing map and then populate it without explicitly taking a lock.

Reply:
It would be good if @donraab could look at this one file, but I suspect you just need to synchronize on existingValue. Also, should it be called `existingValues`?

Reply:
Just checked put(), we synchronize on the new empty collection.

Reply:
refactored and renamed. Also synchronized on existingValues.